interp/interp_example_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp_test
+
+import (
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/interp"
+)
+
+func ExamplePredictor() {
+	xs := []float64{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}
+	ys := []float64{0, 0.001, 0.002, 0.1, 1, 2, 2.5, -10, -10.01, 2.49, 2.53, 2.55}
+
+	var pc interp.PiecewiseConstant
+	var pl interp.PiecewiseLinear
+	var as interp.AkimaSpline
+	var fb interp.FritschButland
+
+	_ = pc.Fit(xs, ys)
```

Comment:
ignoring errors in an example might be an incentive for "bad" code spreading in the wild.

please consider either adding the `3x4` lines of error code checking, or create a slice `[]interp.Fitter{&pc,&pl,&as,&fb}` and apply the fit in a for-loop.